framework/include/physics/PhysicsBase.h

```diff
@@ -96,14 +122,59 @@ class PhysicsBase : public Action
   /// Check that a parameter is set only if the first one is set to true
   void checkSecondParamSetOnlyIfFirstOneTrue(const std::string & param1,
                                              const std::string & param2) const;
```

Comment:
I would move all of these into an `inputparamsutils.h/c`, feels like they are very general and could be reused.

Reply:
There s diamond inheritance problems to deal with because of the need for paramError but I ll see what I can do

Reply:
could consider virtual inheritance

Reply:
I think we should yes. I ll look at it more seriously this could simplify things

but this parameter checking utility that is going away when we have functor-parameter-checking is not the place